prover/src/trace/trace_lde/default/mod.rs

```diff
@@ -207,7 +207,7 @@ where
             positions,
         )];
 
-        // build queries for auxiliary trace segments
+        // build queries for the auxiliary trace segment
         for (i, segment_tree) in self.aux_segment_trees.iter().enumerate() {
             let segment_lde = &self.aux_segment_ldes[i];
             result.push(build_segment_queries(segment_lde, segment_tree, positions));
```

Comment:
Should this be just an if statement? Also, `Self.aux_segment_ldes` and `Self.aux_segment_trees` should probably be just singular objects rather than vectors.

Reply:
Ah yes, I missed those. Fixed. I also renamed `TraceLde::add_aux_segment()` -> `TraceLde::set_aux_segment()`